main/src/ca/uwaterloo/flix/language/phase/unification/BoolUnification.scala

```diff
@@ -51,6 +51,13 @@ object BoolUnification {
       return Ok((Substitution.empty, List(Ast.BroadEqualityConstraint(tpe1, tpe2))))
     }
 
+    // Check for Type.Error
```

Comment:
I think we need to do this recursively in the type. Otherwise we get crashes if there's something nested and we can't translate it.

Reply:
Yikes. That makes this even harder / costly.

Reply:
This seems hard, so I created a ticket: https://github.com/flix/flix/issues/6844